web/docs/internal/gestures/keyman-keyboard-gesture-specs.md

```diff
@@ -0,0 +1,351 @@
+    await-new-timer-elapsed{Time limit between taps has elapsed?}
+    late-new-gesture{New gesture started during modipress hold?}
+    hold-timer-elapsed-subsequent{Tap-hold time limit has elapsed?}
+
+    subgraph Stage1[" "]
+    start --> valid-modipress-check
+    end
+
+    valid-modipress-check -- Yes --> modipress-start
+    valid-modipress-check -- No  --> reject
+
+    subgraph Stage2[" "]
+    reject
+    end
+
+    subgraph Stage3[" "]
+    modipress-start
+    end
+
+    modipress-start -- (emits key) --> modipress-hold
+
+    subgraph Stage4[" "]
+    modipress-hold --> early-new-gesture
+    early-new-gesture -- No --> hold-timer-elapsed-initial
+    end
+
+    early-new-gesture -- Yes --> modipress-end
+    hold-timer-elapsed-initial -- No + input finishes --> modipress-end-multitap-transition
+    hold-timer-elapsed-initial -- Yes --> modipress-end
+
+    subgraph Stage5[" "]
+    modipress-end
+    end
+
+    modipress-end --> complete
+
+    subgraph Stage6[" "]
+    complete
+    end
+
+    subgraph MultitapSupport["Multitap support block"]
+        subgraph Stage7[" "]
+        modipress-end-multitap-transition
+        end
+
+        modipress-end-multitap-transition --> valid-multitap-key
+        valid-multitap-key -- Yes --> modipress-multitap-start
+        valid-multitap-key -- No --> complete
+
+        modipress-multitap-start
+
+        subgraph Stage8[" "]
+        modipress-multitap-start --> await-new-timer-elapsed
+        end
+
+        await-new-timer-elapsed -- No --> modipress-multitap-end
+
+        subgraph Stage9[" "]
+        modipress-multitap-end  --> late-new-gesture
+        late-new-gesture -- No --> hold-timer-elapsed-subsequent
+        end
+
+        late-new-gesture -- Yes --> modipress-multitap-lock-transition
+        hold-timer-elapsed-subsequent -- Yes --> modipress-multitap-lock-transition
+        hold-timer-elapsed-subsequent -- No + input finishes --> modipress-multitap-start
+
+        subgraph Stage10[" "]
+        modipress-multitap-lock-transition
+        end
+
+    end
+
+    await-new-timer-elapsed -- Yes --> complete
+
+    modipress-multitap-lock-transition --> modipress-end
+```
+
+Note that both `modipress-end-multitap-transition` and `modipress-multitap-lock-transition` are transitory (but real!) states that immediately transition to their successors.  They exist to abstractify and clarify the mechanisms at play.
+
+## Flicks
+
+The flick gesture models have their own devoted FSM in order to support their complexities.  Of particular note is that flicks and longpresses compete in a similar space, so the 'flick-start' -> 'flick-mid' transition exists to "lock in" after a small-but-significant amount of motion as a flick, rather than a longpress.  If a key is held too long without enough movement on a key supporting both flicks and longpresses, the longpress FSM will transition first and cancel any further attempt to match a flick for that input.
```

Comment:
```suggestion
The flick gesture models have their own devoted FSM in order to support their complexities.  Of particular note is that flicks and longpresses compete in a similar space, so the 'flick-start' → 'flick-mid' transition exists to "lock in" after a small-but-significant amount of motion as a flick, rather than a longpress.  If a key is held too long without enough movement on a key supporting both flicks and longpresses, the longpress FSM will transition first and cancel any further attempt to match a flick for that input.
```